### PR TITLE
feat: Enrich transactions

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Fixes https://github.com/getsentry/sentry-dotnet/issues/972

### CODE REVIEW

File: src/Sentry/Internal/Hub.cs [hunk 1]
```diff
@@ -473,34 +470,11 @@ public void CaptureTransaction(Transaction transaction, Hint? hint)
...
-                    if (processedTransaction == null)
-                    {
-                        _options.ClientReportRecorder.RecordDiscardedEvent(DiscardReason.EventProcessor, DataCategory.Transaction);
-                        _options.LogInfo("Event dropped by processor {0}", processor.GetType().Name);
-                        return;
-                    }
-                }
-            }
+            var (currentScope, client) = ScopeManager.GetCurrent();
+            // scope ??= currentScope;
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Hub.cs:474]: delete?

File: src/Sentry/Internal/Hub.cs [hunk 2]
```diff
@@ -473,34 +470,11 @@ public void CaptureTransaction(Transaction transaction, Hint? hint)
...
 
-            client.CaptureTransaction(processedTransaction, hint);
+            client.CaptureTransaction(transaction, currentScope, hint);
+            Console.WriteLine("HALP!");
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Hub.cs:477]: we don't want this

**Review by bruno-garcia — COMMENTED** [review]
quick pass

**Review by bruno-garcia — COMMENTED** [review]
quick pass

File: src/Sentry/SentryClient.cs [hunk 3]
```diff
@@ -137,7 +138,33 @@ public void CaptureTransaction(Transaction transaction, Hint? hint)
...
+
+        _options.LogInfo("Capturing transaction.");
+
+        scope.Evaluate();
+        scope.Apply(transaction);
+
+        _enricher.Apply(transaction);
+
+        var processedTransaction = transaction;
+        if (transaction.IsSampled != false)
```
Comment (bruno-garcia) [inline src/Sentry/SentryClient.cs:153]: is this nullable? The fact it's not `!IsSampled` indicates so and it means if value is `null` we'll also go in the block.

I see it's the original code copied but wonder

**Review by bruno-garcia — APPROVED** [review]

File: src/Sentry/SentryClient.cs (hunk 3 above)
Comment (bitsandfoxes) [reply src/Sentry/SentryClient.cs:153]: It's not nullable.

**Pull request merged**